### PR TITLE
Added use case document

### PR DESCRIPTION
**Pull request by Scholars-Mate**
@MelissaHollingshed, I worked on some of the use cases. Please review and comment. I wasn't sure what to have for sponsors, so I left that blank for now.

### CODE REVIEW

File: docs/usecases.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+* Mailing list
+* Travel reimbursement
+
+## Functional
+
+### Authentication
+
+* Administrators and participants should be able to authenticate with an email
+  and password
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Participants can use GitHub OAuth

File: docs/usecases.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+## Administration management
+
+* Administrators should be able to add new administrators
+* Administrators should be able to fetch a list of all administrators
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Administrators can delete Admin

File: docs/usecases.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+* Participants should be able to check schedule
+* Administrators should be able to add events and modify schedule
+
+### Announcements/Notifications
+* Participants should be able to fetch announcements
+* Admins should be able to create/modify/delete announcements
+
+### Sponsors
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Sponsors/Mentors should be able to register with name, skills, a form of contact (slack), and availability on campus 
* Sponsors should be able to add a website link to their company's website, API, or a link to internship opportunities - just one link option

File: docs/usecases.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+### Mailing List
+
+* Administrators should be able to send mass emails to certain participants,
+  e.g. only approved applicants
+* System should be able to mail participants, e.g. application approval, account
+  verification
+* System should be able to mail administrators on specified admin changes
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Send mail to sponsor or mentor contacts with updates

File: docs/usecases.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+### Registration/Applications
+
+* Participants should be able to apply
+* Participants should be able to update/modify applications
+* Participants should be able to check application status
+* Participants should be able to RSVP
+* Administrators should be able to fetch all applications
+* Administrators should be able to approve/decline applications
+* System should be able to verify account/email
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Sponsors/Mentors should be able to register the mentor/mentors they intend to send with email

File: docs/usecases.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+### Check-in
+
+* Administrators should be able to check participants in
+* Administrators should be able to check who is checked in
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Administrators should check in sponsor mentors

File: docs/usecases.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+* Create a use-case diagram
+* Write use case descriptions
+* Write README.md
+* Design the application
+* Design the REST API endpoints (Epic)
+* Design a normalized database schema (Epic)
+  * Design a table for participants
+  * Design a table for administrators
+  * Design a table for applications
+  * Design a table for sponsor info
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Design a table for mentors

File: docs/usecases.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+* Administration management
+* Registration/Applications
+* Check in
+* Participant/meal tracking
+* Schedule/Events
+* Notifications
+* Sponsors
+* Mailing list
+* Travel reimbursement
+
```
Comment (MelissaHollingshed) [inline docs/usecases.md]: * Prizes
      * Administrators can add Prizes and a Description of how to attain the Prize 
      * Administrators can manage by add/edit/delete Prizes
      * Administrators can set Prizes to beginner or standard (boolean)

**Comment by Scholars-Mate** [comment]
Added your suggestions.

**Comment by MelissaHollingshed** [comment]
Can we merge this into the repo in a docs folder?

**Comment by Scholars-Mate** [comment]
@MelissaHollingshed, if everything looks good, you can go ahead and merge.

**Pull request merged**